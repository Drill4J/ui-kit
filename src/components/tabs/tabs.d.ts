import { ReactElement } from 'react';
interface Props {
    className?: string;
    children: ReactElement | ReactElement[];
    activeTab: number | string;
    onSelect: (tabName: string) => void;
}
export declare const TabsPanel: (props: Props) => JSX.Element;
export declare const Tab: DOMComponent<{
    name: string;
}>;
export {};
