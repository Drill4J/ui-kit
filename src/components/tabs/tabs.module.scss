@import '../../common/style-guide/index';

.tabs-container {
  display: flex;

  &__tabs-panel {
    &:not(:last-child) {
      margin-left: 28px;
    }
  }

  &__tab {
    font-size: 14px;
    font-weight: 600;
    line-height: 20px;
    color: map-get($color-palette, blue100);

    &--active {
      pointer-events: none;
      color: map-get($color-palette, black100);
    }
  }
}
