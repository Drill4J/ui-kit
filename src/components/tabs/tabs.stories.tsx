import { useState } from 'react';
import { storiesOf } from '@storybook/react';

import { TabsPanel, Tab } from './tabs';

storiesOf('Tabs', module).add('Tabs', () => {
  const [selectedTab, setSelectedTab] = useState('general');
  return (
    <TabsPanel activeTab={selectedTab} onSelect={setSelectedTab}>
      <Tab name="general">General</Tab>
      <Tab name="system">System</Tab>
      <Tab name="plugins">Plugins</Tab>
    </TabsPanel>
  );
});
