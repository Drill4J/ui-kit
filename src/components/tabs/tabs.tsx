import * as React from 'react';
import { BEM, div } from '@redneckz/react-bem-helper';

import styles from './tabs.module.scss';

interface Props {
  className?: string;
  children: Array<typeof Tab>;
}

const tabsContainer = BEM(styles);

export const TabsContainer = tabsContainer(({ className, children }: Props) => (
  <div className={className}>
    <TabPanel>{children}</TabPanel>
  </div>
));

export const TabPanel = tabsContainer.tabsPanel('div');
export const Tab = tabsContainer.tab(
  div({ onClick: () => {}, active: undefined } as { onClick?: () => void; active?: boolean }),
);
