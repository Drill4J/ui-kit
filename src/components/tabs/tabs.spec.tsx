import React from 'react';
import renderer from 'react-test-renderer';

import { TabsPanel, Tab } from './tabs';

it('renders correctly', () => {
  const tree = renderer.create(
  <TabsPanel activeTab={'general'} onSelect={()=> 'system'}>
    <Tab name="general">General</Tab>
    <Tab name="system">System</Tab>
    <Tab name="plugins">Plugins</Tab>
  </TabsPanel>).toJSON();
  
  expect(tree).toMatchSnapshot();
});