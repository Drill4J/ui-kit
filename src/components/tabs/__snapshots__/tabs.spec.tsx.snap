// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="tabs"
>
  <button
    className="tabs__tab-label tabs__tab-label--active"
    name="general"
    onClick={[Function]}
    type="button"
  >
    General
  </button>
  <button
    className="tabs__tab-label"
    name="system"
    onClick={[Function]}
    type="button"
  >
    System
  </button>
  <button
    className="tabs__tab-label"
    name="plugins"
    onClick={[Function]}
    type="button"
  >
    Plugins
  </button>
</div>
`;
